### ISSUE TITLE
[New documentation] How to hide the instance behind a proxy server at provider that don't care about DMCA requests

### ISSUE THREAD

**Issue by unixfox** [iv-org/documentation#348]
A lot of maintainers have been getting DMCA requests lately (@iv-org/developers please keep this list up-to-date):
- https://github.com/iv-org/documentation/issues/346
- https://github.com/iv-org/documentation/issues/301#issuecomment-1423454175
- https://github.com/iv-org/documentation/issues/241
- me myself
- invidious.namazso.eu @namazso 
- invidious.slipfox.xyz

I've personally dealt with this issue by hosting a proxy server at https://buyvm.net, it's a simple HAProxy for the cost of $3.50/month that redirect all the requests to my original servers.
For now, I have not received any DMCA requests from the provider of my original servers, nor from buyvm.

There are plenty of other DMCA free providers:
- https://lowendtalk.com/discussion/179472/what-are-my-options-for-a-dmca-free-vps
- https://lowendtalk.com/discussion/181443/options-for-dmca-free-vps
- https://lowendtalk.com/discussion/182615/are-there-any-active-dmca-free-vps-offers-from-top-providers

On top of HAProxy, I have deployed a clever technique that redirect the traffic directly to my original servers without going through the proxy server using the Alt-Svc HTTP header: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Alt-Svc
Basically, this mostly only works for real browsers (with HTTP2 & HTTP3), they will see this header and redirect the requests to my original servers that are specified in this HTTP header.
What's great is that this "technique" is not easy to find. I think the dumb DMCA guys only look at the IP address associated with the domain and send the request to that IP address that's it, so they will never think of inspecting the real IP address that a browser is using.

As more and more maintainers are being affected by this issue, I've to document the way I've implemented the ability to hide my servers.

**Comment by unixfox** [comment]
Document created: https://github.com/iv-org/documentation/blob/master/docs/hide-instance-behind-proxy-server.md

**Comment by unixfox** [comment]
Reopening because the 5th step would be to load the bandwidth intensive data (streams, images and more) from another domain in order to reduce the traffic on the proxy.
